Mathlib/Analysis/PSeries.lean

```diff
@@ -19,134 +19,183 @@ converges if and only if so does `∑ k, 2 ^ k f (2 ^ k)`. We prove this test in
+    simp [NNReal.coe_sub (Nat.cast_le (α := ℝ≥0).mpr <| (hu_strict k.lt_succ_self).le)]
+  simp only [nsmul_eq_mul, this]
   apply ENNReal.sum_le_tsum
-#align ennreal.le_tsum_condensed ENNReal.le_tsum_condensed
 
-theorem tsum_condensed_le (hf : ∀ ⦃m n⦄, 1 < m → m ≤ n → f n ≤ f m) :
-    (∑' k : ℕ, 2 ^ k * f (2 ^ k)) ≤ f 1 + 2 * ∑' k, f k := by
-  rw [ENNReal.tsum_eq_iSup_nat' (tendsto_atTop_mono Nat.le_succ tendsto_id), two_mul, ← two_nsmul]
+theorem tsum_schlomilch_le {C : ℕ} (hf : ∀ ⦃m n⦄, 1 < m → m ≤ n → f n ≤ f m) (h_pos : ∀ n, 0 < u n)
+    (h_nonneg : ∀ n, 0 <= f n) (hu_strict : StrictMono u) (h_succ_diff : SuccDiffBounded C u) :
+    ∑' k : ℕ, (u (k + 1) - u k) * f (u k) ≤ (u 1 - u 0) * f (u 0) + C * ∑' k, f k := by
+  have hu : Monotone u := by
+    apply StrictMono.monotone hu_strict
+  rw [ENNReal.tsum_eq_iSup_nat' (tendsto_atTop_mono Nat.le_succ tendsto_id)]
   refine'
     iSup_le fun n =>
       le_trans _
         (add_le_add_left
-          (nsmul_le_nsmul_right (ENNReal.sum_le_tsum <| Finset.Ico 2 (2 ^ n + 1)) _) _)
-  simpa using Finset.sum_condensed_le hf n
-#align ennreal.tsum_condensed_le ENNReal.tsum_condensed_le
-
+          (mul_le_mul_of_nonneg_left (ENNReal.sum_le_tsum <| Finset.Ico (u 0 + 1) (u n + 1)) _) _)
+  simpa using Finset.sum_schlomilch_le hf h_pos h_nonneg hu h_succ_diff n
+  apply zero_le _
 end ENNReal
 
 namespace NNReal
 
+open BigOperators Finset
+
 open ENNReal in
-/-- Cauchy condensation test for a series of `NNReal` version. -/
-theorem summable_condensed_iff {f : ℕ → ℝ≥0} (hf : ∀ ⦃m n⦄, 0 < m → m ≤ n → f n ≤ f m) :
-    (Summable fun k : ℕ => (2 : ℝ≥0) ^ k * f (2 ^ k)) ↔ Summable f := by
-  simp only [← ENNReal.tsum_coe_ne_top_iff_summable, Ne.def, not_iff_not, ENNReal.coe_mul,
-    ENNReal.coe_pow, ENNReal.coe_two]
+/-- for a series of `NNReal` version. -/
+theorem summable_schlomilch_iff {C : ℕ} {u : ℕ → ℕ} {f : ℕ → ℝ≥0}
+    (hf : ∀ ⦃m n⦄, 0 < m → m ≤ n → f n ≤ f m)
+    (h_pos : ∀ n, 0 < u n) (h_nonneg : ∀ n, 0 <= f n) (hu_strict : StrictMono u) (hCpos : C > 0)
+    (h_succ_diff : SuccDiffBounded C u) :
+    (Summable fun k : ℕ => (u (k + 1) - (u k : ℝ≥0)) * f (u k)) ↔ Summable f := by
+  simp only [← ENNReal.tsum_coe_ne_top_iff_summable, Ne.def, not_iff_not, ENNReal.coe_mul]
   constructor <;> intro h
   · replace hf : ∀ m n, 1 < m → m ≤ n → (f n : ℝ≥0∞) ≤ f m := fun m n hm hmn =>
       ENNReal.coe_le_coe.2 (hf (zero_lt_one.trans hm) hmn)
-    simpa [h, ENNReal.add_eq_top, ENNReal.mul_eq_top] using ENNReal.tsum_condensed_le hf
+    replace h_nonneg : ∀ n, 0 ≤ (f n : ℝ≥0∞) := fun n =>
+      ENNReal.coe_le_coe.2 (h_nonneg n)
+    obtain hC := ENNReal.tsum_schlomilch_le hf h_pos h_nonneg hu_strict h_succ_diff
+    have C_nonzero : C ≠ 0 := ne_of_gt hCpos
+    have : (↑(u 1) - ↑(u 0)) * ↑(f (u 0)) + ↑C * ∑' (k : ℕ), ↑(f k) = ∞ := by exact eq_top_mono hC h
+    simpa [ENNReal.add_eq_top, ENNReal.mul_ne_top, ENNReal.mul_eq_top, C_nonzero]
```

Comment:
```suggestion
    replace h_nonneg : ∀ n, 0 ≤ (f n : ℝ≥0∞) := fun n =>
      ENNReal.coe_le_coe.2 (h_nonneg n)
    obtain hC := tsum_schlomilch_le hf h_pos h_nonneg hu_strict h_succ_diff
    have C_nonzero : C ≠ 0 := ne_of_gt hCpos
    have : (↑(u 1) - ↑(u 0)) * ↑(f (u 0)) + ↑C * ∑' (k : ℕ), ↑(f k) = ∞ := by exact eq_top_mono hC h
    simpa [add_eq_top, mul_ne_top, mul_eq_top, C_nonzero]
```
since `ENNReal` is opened I don't see why we should qualify the names (`coe_le_coe` needs to be qualified however)